lib/index.js

```diff
@@ -373,37 +378,23 @@ MaplibreGeocoder.prototype = {
 
     // ENTER
     if (e.keyCode === 13) {
-      if (this.options.showResultMarkers) {
-        if (this._typeahead.data.length < 1) return;
-
-        var results = this._typeahead.data.slice(0, this.options.limit);
-
-        this._clearEl.style.display = "none";
-
-        if (this.options.flyTo && this._maplibregl) {
-          if (this._map) {
-            var defaultFlyOptions = { padding: 100 };
-            var flyOptions = extend({}, defaultFlyOptions, this.options.flyTo);
-            var bounds = new this._maplibregl.LngLatBounds();
-            results.forEach(function (feature) {
-              bounds.extend(feature.geometry.coordinates);
-            });
-
-            this._map.fitBounds(bounds, flyOptions);
-          }
-        }
-
-        if (results.length > 0 && this._maplibregl) {
-          this._handleResultMarkers(results);
+      if (!this.options.showResultsWhileTyping) {
+        this._geocode(target.value);
+      } else {
+        if (this.options.showResultMarkers) {
+          this._fitBoundsForMarkers();
         }
```

Comment:
Lacking some context on this change here, why `this._fitBoundsForMarkers()` is not applicable here when `showResultsWhileTyping==false` and rather moved to lines 641,642.

Reply:
With the new feature when `showResultsWhileTyping` is false we will only show results after pressing enter. In that case we also only want to attempt to `_fitBoundsForMarkers()` when showing results which is done in the `_geocode()` method (lines 641,642).

When `showResultsWhileTyping` is true then `geocode()` method is being automatically called while typing, already showing results in the field. In that case when users press enter they are either pressing enter on the text field to indicate they want to `_fitBoundsForMarkers()` or pressing enter (or clicking on) a specific result indicating they want to `flyTo` a single marker